packages/frontend/src/components/DashboardFilter/FilterConfiguration/TileFilterConfiguration.tsx

```diff
@@ -93,10 +93,21 @@ const TileFilterConfiguration: FC<Props> = ({
     const tileTargetList = useMemo(() => {
         return sortedTileWithFilters.map(([tileUuid, filters], index) => {
             const tile = tiles.find((t) => t.uuid === tileUuid);
+
+            // tileConfig overrides the default filter state for a tile
+            // if it is there, we set the selected field to its
+            // stored value. If not, we use the field default
             const tileConfig = filterRule.tileTargets?.[tileUuid];
+            const fieldId = tileConfig?.fieldId || getFieldId(field);
+            const selectedField = filters?.find((f) => {
+                return getFieldId(f) === fieldId;
+            });
 
-            const isFilterAvailable =
-                filters?.some(matchFieldByType(field)) ?? false;
```

Comment:
hey @magnew , I am not sure why this was removed and logic seems little bit off.

here is how the logic worked (/supposed to work) before:

- if you pick `order.date` field and tile explore has order model then it should automatically select `order.date` field in the UI and automatically apply that filter to the tile
- if you pick `order.date` field and tile **does not** have order model but has `customer.created_date` then UI should still show input as available and customer can set an overriding field - altho it should not override automatically to another field with the same type
- if you pick `order.date` field and tile **does not** have anything in their explore with date type then the input should be disabled and override should not exist/should not be possible - filter should not apply to the tile

in the `filterApplies` logic I don't see where we check that

```
    (!!selectedField && tileConfig?.fieldId !== '') ||
                !!tileConfig?.fieldId;
```

to help with the logic I created `matchFieldExact` `matchFieldByTypeAndName` and `matchFieldByType` filters which does not seem to be utilized anymore

Reply:
Oh, I remember. It looks unnecessary since the array we are looping through (sortedTileWithFilters) is already filtered using those functions. FWIW, it does seem to work how you described. Did you find a case where it doesn't? 

I will triple check.

Reply:
Ok, @IrakliJani, thank you! I was wrong. I dug in again and I see what you're saying. I think I fixed it.